knime_extension/src/nodes/spatialnetwork.py

```diff
@@ -868,7 +878,292 @@ def ckdnearest(gdA, gdB):
+        ):
+            # generate nodes and edges geodataframes from graph
+            primal = True
+            for nid, n in enumerate(net):
+                net.nodes[n][nodeID] = nid
+            return primal_to_gdf(
+                net,
+                points=points,
+                lines=lines,
+                spatial_weights=spatial_weights,
+                nodeID=nodeID,
+            )
+
+        def points_to_gdf(net):
+            node_xy, node_data = zip(*net.nodes(data=True))
+            if isinstance(node_xy[0], int) and "x" in node_data[0].keys():
+                geometry = [
+                    Point(data["x"], data["y"]) for data in node_data
+                ]  # osmnx graph
+            else:
+                geometry = [Point(*p) for p in node_xy]
+            gdf_nodes = gp.GeoDataFrame(list(node_data), geometry=geometry)
+            if "crs" in net.graph.keys():
+                gdf_nodes.crs = net.graph["crs"]
+            return gdf_nodes
+
+        def lines_to_gdf(net, points, nodeID):
+            starts, ends, edge_data = zip(*net.edges(data=True))
+            gdf_edges = gp.GeoDataFrame(list(edge_data))
+
+            if points is True:
+                node_start = []
+                node_end = []
+                for s in starts:
+                    node_start.append(net.nodes[s][nodeID])
+                for e in ends:
+                    node_end.append(net.nodes[e][nodeID])
+                gdf_edges["node_start"] = node_start
+                gdf_edges["node_end"] = node_end
+
+            if "crs" in net.graph.keys():
+                gdf_edges.crs = net.graph["crs"]
+
+            return gdf_edges
+
+        # End----------------Repeated code as those in Street Network Matrix
+
+        def gdf_to_osmgraph(gdf):
+            graph = gdf_to_nx(gdf)
+            nodes, edges = nx_to_gdf(
+                graph, points=True, lines=True, spatial_weights=False
+            )
+            nodes["x"] = nodes.geometry.x
+            nodes["y"] = nodes.geometry.y
+            nodes = nodes.rename(columns={"nodeID": "osmid"})
+            edges = edges.rename(columns={"node_start": "u", "node_end": "v"})
+            edges["k"] = list(range(1, (edges.shape[0] + 1)))
+            edges = edges.set_index(["u", "v", "k"])
+            G = ox.utils_graph.graph_from_gdfs(nodes, edges)
+            return G
+
+        # Cross join Data
+        c_gdf = input1.to_pandas().rename(columns={self.c_geo_col: "geometry"})
+        r_gdf = input2.to_pandas().rename(columns={self.r_geo_col: "geometry"})
+        c_gdf = gp.GeoDataFrame(c_gdf, geometry="geometry")
+        r_gdf = gp.GeoDataFrame(r_gdf, geometry="geometry")
+        r_gdf = r_gdf.rename(columns={self.r_cost_col: "time"})
+        # Set a lat\Lon CRS
+
+        crsinput = CRS.from_user_input(r_gdf.crs)
+        if crsinput.is_geographic:
+            logging.warning("Unit as Degree, Please use Projected CRS")
```

Comment:
Maybe change to:
_Input CRS has degree unit. Use default projected CRS epsg:3857_

Reply:
revised